### PR TITLE
#BN-24:Ch set continous pipeline and integration 

### PR DESCRIPTION
**Pull request by Niyobuzima**
# PR Description
Please include a brief description of the pull request you want to raise.
# Description of tasks that were expected to be completed
List all the tasks and sub-tasks that were assigned to you Make sure you check every completed task. list them in this style
- [x] Configured Testing environment with either Mocha, Chai and Sinon, or Jest and Supertest to ensure that all functionalities are tested before deployment.
- [x] GitHub Actions is configured as a continuous integration service used to build and test the app.
- [x] Configured a deployment pipeline on Cyclic or Render, or any other host
- [x] GitHub Actions  badge has been added to your repo
- [x] Configured CodeClimate for code analysis and Maintainability checking.
# Description of tasks that were expected to be completed and that are not yet completed
# How this may be tested?
Please describe the tests that you ran to verify your functionalities. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.
1. run `npm i` to install all required dependencies
2. run `npm run dev`
# Screenshots (If appropriate/applicable)
Provide screenshots related to your PR in case it's appropriate and applicable
# Please check this Checklist before you submit your PR:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My code generates no warnings
- [x] There are no vulnerabilities

### CODE REVIEW

File: .codeclimate.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  method-lines:
+    enabled: false
+  nested-control-flow:
+    enabled: false
+  return-statements:
+    enabled: false
+  similar-code:
+    enabled: false
+  identical-code:
+    enabled: false
```
Comment (dushimeemma) [inline .codeclimate.yml]: add a line at the end of this file @Niyobuzima

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+          restore-keys: |
+            ${{ runner.os }}-node-
+      - name: Install dependencies
+        run: npm install
+
+      - name: Run linter
+        run: npm run lint
+
+      - name: Run tests
+        run: npm test
```
Comment (dushimeemma) [inline .github/workflows/CI.yml]: add a line at the end of this file

File: src/controllers/userController.js [hunk 3]
```diff
@@ -1,91 +1,296 @@
...
+      data: User,
+    });
+  } catch (err) {
+    return res.status(500).json({
+      status: 500,
+      message: req.t("server_error"),
+      Error: err.message,
+    });
+  }
+}
```
Comment (dushimeemma) [inline src/controllers/userController.js]: add a line at the end of this file

File: src/database/config/config.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    port: process.env.DB_PROD_PORT,
+    sslmode: require,
+    dialectOptions: {
+      ssl: {
+        require: true,
+        rejectUnauthorized: false,
+      },
+    },
+  },
+};
```
Comment (dushimeemma) [inline src/database/config/config.js:44]: add a line at the end of this file

File: src/tests/sum.test.js
Comment (dushimeemma) [inline src/tests/sum.test.js:1]: I don't understand the reason of this test file

**Review by uwdjany — APPROVED** [review]
well done!

File: src/tests/sum.test.js
Comment (Niyobuzima) [reply src/tests/sum.test.js:1]: this was a dummy test that was showing how the test will be done

Comment (dushimeemma) [inline src/tests/sum.test.js:1]: Use project related test cases
Example:

- Test `/` endpoint

**Review by dushimeemma — APPROVED** [review]
Good job 🥇

**Pull request merged**